src/CheckoutSdk/ApiClient.cs

```diff
@@ -251,14 +253,37 @@ private async Task<TResult> DeserializeJsonAsync<TResult>(HttpResponseMessage ht
 
         private async Task<dynamic> DeserializeJsonAsync(HttpResponseMessage httpResponse, Type resultType)
         {
-            if (httpResponse.StatusCode == HttpStatusCode.NoContent ||
-                httpResponse.Content.Headers.ContentLength == 0)
+            dynamic deserializedObject;
+            if (httpResponse.StatusCode == HttpStatusCode.NoContent || httpResponse.Content.Headers.ContentLength == 0)
+            {
+                deserializedObject = Activator.CreateInstance(resultType);
+            }
+            else
             {
-                return null;
+                var json = await httpResponse.Content.ReadAsStringAsync();
+                deserializedObject = _serializer.Deserialize(json, resultType);
             }
 
-            var json = await httpResponse.Content.ReadAsStringAsync();
-            return _serializer.Deserialize(json, resultType);
+            await TransformHttpMetadata(httpResponse, resultType, deserializedObject);
+
+            return deserializedObject;
+        }
+
+        private static async Task TransformHttpMetadata(HttpResponseMessage httpResponse, Type resultType,
+            dynamic deserializedObject)
+        {
+            if (resultType.IsSubclassOf(typeof(HttpMetadata)))
+            {
+                ((HttpMetadata)deserializedObject).Body = await httpResponse.Content.ReadAsStringAsync();
+                ((HttpMetadata)deserializedObject).HttpStatusCode = (int)httpResponse.StatusCode;
+                IDictionary<string, string> headers = new Dictionary<string, string>();
+                foreach (var header in httpResponse.Headers)
```

Comment:
It's not possible to set the IDictionary directly?

Reply:
The wrapper of "httpResponse.Headers" is a KeyValuePair<string,IEnumerable<>> I thought that was better to provide only a Dictionary of <string,string>